File: Fibonacci Sequence/Fibonacci.go
package main

import "fmt"

// Find the nth number of Fibonacci sequence
// Fibonacci sequence: 0, 1, 1, 2, 3, 5, 8, 13, 21, ...
func findNthFibonacciNumber(n int) int64 {
	if n == 0 || n == 1 {
		return int64(n)
	}
	var firstPrv int64 = 0
	var secondPrv int64 = 1

	for i := 2; i <= n; i++ {
		var tmp int64 = firstPrv + secondPrv
		firstPrv = secondPrv
		secondPrv = tmp
	}
	return secondPrv
}

func main() {
	var idxTarget int

	fmt.Printf("Enter the required index of Fibonacci sequence: ")
	fmt.Scanf("%d", &idxTarget)

	fmt.Printf("The %dth element of Fibonacci sequence is %d\n", idxTarget, findNthFibonacciNumber(idxTarget))
}

File: Fibonacci Sequence/fibonacci.c
#include <stdio.h>

typedef unsigned long long U64;

U64 fibonacci(U64 n);

// print the nth element of the fibonacci sequence. n should be a non-negative integer
int main()
{
    U64 n;
    printf("Please enter a non-negative integer: ");
    scanf("%llu", &n);
    // fibonacci(3);

    printf(
        "%lluth element of fibonacci sequence = %llu\n",
        n,
        fibonacci(n)
    );

    return 0;
}

/**
 * return the nth element of the fibonacci sequence. n should be a non-negative integer
 *
 * @param n the element to find
 * @return the nth element of the fibonacci sequnce
 */
U64 fibonacci(U64 n)
{
    if (n == 0) {
        return 0;
    }
    else if (n == 1) {
        return 1;
    }
    else {
        U64 t1 = 0;
        U64 t2 = 1;

        // TODO: check for integer overflow
        U64 result, temp;
        for (U64 i = 1; i <= n - 1; ++i) {
            result = t1 + t2;

            temp = t2;
            t2 = result;
            t1 = temp;
        }

        return result;
    }
}
